test/framework/clusterctl/repository.go

```diff
@@ -149,8 +149,13 @@ func YAMLForComponentSource(ctx context.Context, source ProviderVersionSource) (
 		}
 		data = buf
 	case KustomizeSource:
+		// Set Path of kustomize binary using CAPI_KUSTOMIZE_PATH env
+		kustomizePath, ok := os.LookupEnv("CAPI_KUSTOMIZE_PATH")
+		if !ok {
+			kustomizePath = "kustomize"
```

Comment:
Q: should this be `hack/tools/bin/kustomize` (assuming the current folder is the project root folder) so we use the exact version CAPI E2E expects?

@sbueringer opinions?

Note, this is an improvement on top of the original scope of this PR, so I'm fine also with moving it to a separated PR/issue

Reply:
Isn't the test framework also imported and used in other repositories? If that's the case, we can't rely on the Makefile always generating the Kustomize binary in the same format, although we could fallback?

Reply:
I'm not sure if we should change anything (apart from the current changes).

As Vince mentioned we can't really assume that other providers have the kustomize binary at the same path. Implementing a fallback also wouldn't really make sure that kustomize from that path is used. 

Even if kustomize from that path is used, there are currently no validations which version of kustomize is stored there. We also can't really add a version check for a specific version as different providers are using different versions of kustomize (saw even major version differences in the past (kustomize v3.x vs. v4.x).

In our e2e script today we ensure that the right kustomize is used via:
```bash
export PATH="${REPO_ROOT}/hack/tools/bin:${PATH}"
```

If we want to be really sure that in our e2e tests the right version is used we could probably set the `CAPI_KUSTOMIZE_PATH` var in `docker.yaml`.